connexion/decorators/security.py

```diff
@@ -53,104 +94,131 @@ def security_passthrough(function):
+            val = request.query.get(name)
+            if val is not None:
+                return func(val, required_scopes)
+        elif loc == 'header':
+            val = request.headers.get(name)
+            if val is not None:
+                return func(val, required_scopes)
 
-def validate_token_info(token_info, allowed_scopes):
+    logger.info("... No auth provided. Aborting with 401.")
+    raise OAuthProblem(description='No authorization token provided')
+
+
+def validate_scope(required_scopes, token_scopes):
     """
-    :param allowed_scopes:
-    :param token_info: Dictionary containing the token_info
-    :type token_info: dict
-    :return: None
+    :param required_scopes: Scopes required to access operation
+    :param token_scopes: Scopes granted by authorization server
+    :rtype: bool
     """
-    scope = token_info.get('scope') or token_info.get('scopes')
-    if isinstance(scope, list):
-        user_scopes = set(scope)
+    required_scopes = set(required_scopes)
+    if isinstance(token_scopes, list):
+        token_scopes = set(token_scopes)
     else:
-        user_scopes = set(scope.split())
-    logger.debug("... Scopes required: %s", allowed_scopes)
-    logger.debug("... User scopes: %s", user_scopes)
-    if not allowed_scopes <= user_scopes:
+        token_scopes = set(token_scopes.split())
+    logger.debug("... Scopes required: %s", required_scopes)
+    logger.debug("... Token scopes: %s", token_scopes)
+    if required_scopes > token_scopes:
         logger.info(textwrap.dedent("""
-                    ... User scopes (%s) do not match the scopes necessary to call endpoint (%s).
+                    ... Token scopes (%s) do not match the scopes necessary to call endpoint (%s).
                      Aborting with 403.""").replace('\n', ''),
-                    user_scopes, allowed_scopes)
-        raise OAuthScopeProblem(
-            description='Provided token doesn\'t have the required scope',
-            required_scopes=allowed_scopes,
-            token_scopes=user_scopes
-        )
-    logger.info("... Token authenticated.")
+                    token_scopes, required_scopes)
+        return False
+    return True
 
+def verify_oauth(token_info_func, scope_validate_func):
+    def wrapper(token, required_scopes):
+        token_info = token_info_func(token)
+        if token_info is None:
+            raise OAuthProblem(
+                description='Provided oauth token is not valid'
+            )
 
-def verify_oauth_local(token_info_func, allowed_scopes, function):
-    """
-    Decorator to verify oauth locally
+        # Fallback to 'scopes' for backward compability
+        token_scopes = token_info.get('scope', token_info.get('scopes', ''))
+        if not scope_validate_func(required_scopes, token_scopes):
+            raise OAuthScopeProblem(
+                description='Provided token doesn\'t have the required scope',
+                required_scopes=oauth_args[2],
```

Comment:
I don't think `oauth_args` is in scope here.
I think you want `required_scopes`